racket/collects/racket/hash-code.rkt

```diff
@@ -0,0 +1,54 @@
+#lang racket/base
+
+(provide hash-code-combine hash-code-combine-unordered)
+
+(require racket/fixnum)
+
+;; unsigned integer view of -2, not a fixnum
+(define (um2) (arithmetic-shift (most-positive-fixnum) 1))
+
+;; fixnum shift-left logical with wraparound
+(define fxsll/wraparound fxlshift/wraparound)
+
+;; fixnum logical exclusive or
+(define fxlogxor fxxor)
+
+;; fixnum shift-right logical
+;; shifting negative fixnums can produce large positive fixnums
+(define (fxsrl a b)
+  (cond
+    [(and (negative? a) (positive? b))
+     (arithmetic-shift (bitwise-and a (um2)) (- b))]
+    [else (fxrshift a b)]))
+
+;; Adapted from racket/src/cs/rumble/hash-code.ss
+;; which is adapted from ChezScheme/s/newhash.ss
+
+;; maps non-fixnum integers to positive fixnums
+(define (->fx v)
+  (if (fixnum? v)
+      v
+      (bitwise-and v (most-positive-fixnum))))
+
+;; Adapted from `mix-hash-code` in racket/src/cs/rumble/hash-code.ss
+;; which is adapted from an `update` function in ChezScheme/s/newhash.ss
+;; and adapted from One-at-a-Time Hash in "A Hash Function for Hash Table Lookup"
+;; Bob Jenkins's expansion of his September 1997 Dr Dobbs article on Hash Functions
+;; http://www.burtleburtle.net/bob/hash/doobs.html#one
+(define (fxmix-hash-code hc)
+  (let ([hc2 (fx+/wraparound hc (fxsll/wraparound (fx+/wraparound hc 1) 10))])
+    (fxlogxor hc2 (fxsrl hc2 6))))
+
+;; NOTE: this `hash-code-combine` function is NOT the same as the
+;; `hash-code-combine` from racket/src/cs/rumble/hash-code.ss
```

Comment:
This is no longer accurate, since you also change `racket/src/cs/rumble/hash-code.ss`, correct?